### DISCUSSION TITLE
How to turn off initial focus when opening headlessui?

### DISCUSSION THREAD

**Discussion by youngseo-im** [tailwindlabs/headlessui#2549] [Help]
I'm using headless ui in my vue3 + typescript project

How can I turn off the initial focus on a slotted component input?

Assigning a null value to initialFocus did not work.

`
<script setup lang="ts">
import { Dialog, DialogOverlay, TransitionRoot, TransitionChild, DialogPanel } from '@headlessui/vue'
import { ref } from 'vue'

const props = defineProps<{
  isOpen: boolean
  closeModal: () => void
}>()

const initialFocus = ref<HTMLElement | null>(null)
</script>
<template>
  <div>
    <Teleport to="body">
      <TransitionRoot :show="props.isOpen">
        <Dialog as="div" class="fixed inset-0 overflow-y-auto" :initial-focus="initialFocus" @close="closeModal">
          <div class="flex items-end justify-center min-h-screen text-center">
            <TransitionChild
              as="div"
              class="fixed inset-0 transition-opacity"
              enter-from-class="opacity-0"
              enter-to-class="opacity-100"
              leave-from-class="opacity-100"
              leave-to-class="opacity-0">
              <DialogOverlay class="fixed inset-0 bg-black opacity-30"></DialogOverlay>
            </TransitionChild>

            <TransitionChild
              as="div"
              enter-from-class="translate-y-4 opacity-0 sm:translate-y-0 sm:scale-95"
              enter-to-class="translate-y-0 opacity-100 sm:scale-100"
              leave-from-class="translate-y-0 opacity-100 sm:scale-100"
              leave-to-class="translate-y-4 opacity-0 sm:translate-y-0 sm:scale-95"
              class="w-full">
              <DialogPanel
                class="w-full px-48 pt-40 pb-32 align-middle transition-all transform bg-white shadow-xl rounded-tl-48 rounded-tr-48">
                <slot></slot>
              </DialogPanel>
            </TransitionChild>
          </div>
        </Dialog>
      </TransitionRoot>
    </Teleport>
  </div>
</template>
`

![스크린샷 2023-06-21 13 36 47](https://github.com/tailwindlabs/headlessui/assets/75972718/ee45e841-5f12-4183-a451-e5293a87ae05)

**Comment by 1mehdifaraji** [comment]
For accessibility reasons, dialog should contain at least one focusable element. By default, the Dialog component will focus the first focusable element (by DOM order) once it is rendered, and pressing the Tab key will cycle through all additional focusable elements within the contents.

To disable this behavior or modify it first you need to define a react ref state:
```javascript
const dialogRef = useRef(null);
```
Then define `initialFocus` prop of the `Dialog` component to it and have a `hidden` `div` for your `ref` state.
```javascript
<Dialog initialFocus={dialogRef ? dialogRef : undefined}>
    <div style={{ display: "none" }} ref={dialogRef} />
</Dialog>
```
You can also set the element you want to be focused at first render like this:
```javascript
<Dialog.Panel>
        <Dialog.Title>...</Dialog.Title> 
        <button ref={dialogRef}>
            ....
        </button>
</Dialog.Panel>
```

Vue code defining `initialFocus`:
```javascript
const completeButtonRef = ref(null)

<Dialog :initialFocus="dialogRef">
   <div ref="dialogRef" :style="{display : 'none' }" /> 
</Dialog>
```
```javascript
<button ref="dialogRef">
   ...
</button>
```
